compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -684,6 +684,9 @@ class ExpressionCodegen(
 
     override fun visitWhen(expression: IrWhen, data: BlockInfo): StackValue {
         expression.markLineNumber(startOffset = true)
+        val switchGen = SwitchGenerator(expression, data, this)
+        if (switchGen.match())
```

Comment:
I'd somewhat prefer if the public API would be something like this:
```
val switch = SwitchGenerator(expression, data, this).generate()
if (switch != null) return switch
```

In this case, you wouldn't need mutable fields in `SwitchGenerator` to retain information between `match` and `gen`; the logic in `match` could be inlined into `gen` with local variables holding things like `thenExpressions ` and `elseExpression`

Reply:
Done.